### PR TITLE
Refactor collector package tests

### PR DESCRIPTION
**Pull request by arbulu89**
Refactor collector package tests

### CODE REVIEW

File: internal/discovery/collector/client_test.go [hunk 1]
```diff
@@ -24,12 +25,15 @@ func TestCollectorClientTestSuite(t *testing.T) {
 }
 
 func (suite *CollectorClientTestSuite) TestCollectorClientPublishingSuccess() {
-	collectorClient := NewCollectorClient(
-		&Config{
+	httpClient := http.DefaultClient
```
Comment (nelsonkopliku) [inline internal/discovery/collector/client_test.go:28]: Not a biggie, however what about having `httpClient` set in the suite and pass it as the second argument of `NewCollectorClient`?
```go
collectorClient := collector.NewCollectorClient(
    &collector.Config{/** ... */},
    suite.httpClient,
)
```
as in publishing_test.go?

Comment (arbulu89) [reply internal/discovery/collector/client_test.go:28]: No problem! I can change that

Comment (arbulu89) [reply internal/discovery/collector/client_test.go:30]: Done

**Review by nelsonkopliku — APPROVED** [review]

**Review by EMaksy — COMMENTED** [review]
Looks good :)

**Review by CDimonaco — APPROVED** [review]

**Pull request merged**